lib/inferno/dsl/fhir_evaluation/rules/all_references_resolve.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllReferencesResolve < Rule
+          def check(context)
+            # resource_type_ids is for quick look up when there is a reference type
+            # resource_ids is for quick look up when there is no type (i.e. uuid used)
+            util = Inferno::Utils::EvaluatorUtil
+            _, resource_type_ids, resource_ids, references = util.extract_ids_references(context.data)
+            unresolved_references = Hash.new { |h, k| h[k] = [] }
+            references.each do |k, v|
```

Comment:
`k` and `v` should have meaningful names here.

Reply:
Addressed